app/src/app/components/contents/common/DataTable.tsx

```diff
@@ -1,47 +1,74 @@
-import { ColumnDef, flexRender, getCoreRowModel, useReactTable } from "@tanstack/react-table";
+import { ColumnDef, flexRender, getCoreRowModel, getPaginationRowModel, useReactTable } from "@tanstack/react-table";
 import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from "../../Base/Table";
+import { Pagination } from "./Pagination";
+import { PAGE_SIZE } from "../../../../lib/constants";
 
 interface DataTableProps<TData> {
   columns: ColumnDef<TData, any>[];
   data: TData[];
+  enablePagination?: boolean;
 }
 
-export const DataTable = <TData,>({ data, columns }: DataTableProps<TData>) => {
-  const table = useReactTable({ columns, data, getCoreRowModel: getCoreRowModel() });
+export const DataTable = <TData,>({ data, columns, enablePagination }: DataTableProps<TData>) => {
+  const table = useReactTable({
+    columns,
+    data,
+    getCoreRowModel: getCoreRowModel(),
+    ...(enablePagination && { getPaginationRowModel: getPaginationRowModel() }),
+    initialState: {
+      pagination: {
+        pageSize: PAGE_SIZE
```

Comment:
I think we might eventually want to support different page sizes for different entities - 50 packets might be about right, but might be too many for roles. But don't need to do anything about that now..

Reply:
you are right i have made it so we pass in the page size here now :smile: